Sources/Core/BasePresenter.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  BasePresenter.swift
+//  Ahoy ( https://github.com/xmartlabs/Xniffer)
+//
+//  Copyright (c) 2017 Xmartlabs ( http://xmartlabs.com )
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+import UIKit
+
+open class BasePresenter: OnboardingPresenter {
+
+    public typealias OnboardingData = (titleText: String, bodyText: String, image: UIImage?)
```

Comment:
Maybe you can make `OnboardingData` an `associatedtype` of `OnboardingPresenter`. This way a developer will be able to use their custom data structure for the information shown in the onboarding screen. It's not hard to implement and I think it is a good feature. What do you think?